pkg/netstore/netstore.go

```diff
@@ -74,10 +102,28 @@ func (s *store) Get(ctx context.Context, mode storage.ModeGet, addr swarm.Addres
 // returns a storage.ErrInvalidChunk error when
 // encountering an invalid chunk.
 func (s *store) Put(ctx context.Context, mode storage.ModePut, chs ...swarm.Chunk) (exist []bool, err error) {
+	var chunks []swarm.Chunk
 	for _, ch := range chs {
 		if !s.validator.Validate(ch) {
 			return nil, storage.ErrInvalidChunk
 		}
+
+		// TODO: his double validate in case of a SOC chunk should be avoided in future by
```

Comment:
no  future sorry, i really cannot afford technical debt like this going  in the code for a not so important feature,, especially when it is not needed.
put the callback within the validator or make the validator return type and condition on that etc

Reply:
I have suggested that validator is not changed in this pr. It is already getting large and validator change should add even more changes across the codebase. I think that it should be done in a different pr, before or after this pr is merged.

Reply:
Changed the Validator interface to expose chunk type also.